src/lib/pubkey/pk_keys.h

```diff
@@ -207,6 +207,18 @@ class BOTAN_PUBLIC_API(2,0) Public_Key : public virtual Asymmetric_Key
       virtual std::unique_ptr<PK_Ops::Verification>
          create_verification_op(const std::string& params,
                                 const std::string& provider) const;
+
+      /**
+      * This is an internal library function exposed on key types.
+      * In all cases applications should use wrappers in pubkey.h
+      *
+      * Return a verification operation for this key/params or throw
+      * @param signature_algorithm the X509 algorithm identifier
+      * @param provider the provider to use
+      */
```

Comment:
I think this deserves a few more details. Maybe:

```suggestion
      /**
      * This is an internal library function exposed on key types.
      * In all cases applications should use wrappers in pubkey.h
      *
      * Return a verification operation for this combination of key and
      * signature algorithm or throw. @param signature_algorithm is the
      * X.509 algorithm identifier encoding the padding scheme and hash
      * hash function used in the signature if applicable.
      *
      * @param signature_algorithm the X509 algorithm identifier
      * @param provider the provider to use
      */
```

Reply:
:+1: